src/main/java/frc/robot/subsystems/Elevator/ElevatorIOSim.java

```diff
@@ -0,0 +1,133 @@
+import edu.wpi.first.wpilibj.motorcontrol.PWMTalonFX;
+import edu.wpi.first.wpilibj.simulation.BatterySim;
+import edu.wpi.first.wpilibj.simulation.DCMotorSim;
+import edu.wpi.first.wpilibj.simulation.ElevatorSim;
+import edu.wpi.first.wpilibj.simulation.EncoderSim;
+import edu.wpi.first.wpilibj.simulation.PWMSim;
+import edu.wpi.first.wpilibj.simulation.RoboRioSim;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants.WoodbotConstants;
+import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismLigament2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismRoot2d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj.util.Color8Bit;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+
+public class ElevatorIOSim implements ElevatorIO {
+  /** Creates a new ElevatorIOSim. */
+
+  private DCMotor gearbox = DCMotor.getFalcon500(1);
+  public static Encoder encoder = new Encoder(0, 1);
+
+  final double UPPER_LIMIT = 2;
+  final double LOWER_LIMIT = 0;
+
+  final double kA = 0.0;
+  final double kD = 0.35;
+  final double kG = 0.65;
+  final double kI = 0.2;
+  final double kP = 0.55;
+  final double kS = 0.05;
+  final double kV = 0.0;
+
+  private ProfiledPIDController pidController = new ProfiledPIDController(kP, kI, kD,
+      new TrapezoidProfile.Constraints(600, 300));
+  private ElevatorFeedforward feedforward = new ElevatorFeedforward(kS, kG, kV, kA);
+
+  private final PWMTalonFX motor = new PWMTalonFX(1);
+
+  public final ElevatorSim elevatorSim = new ElevatorSim(
+      gearbox,
+      1.0, // elevator gearing
+      2.0, // carriage mass
+      Units.inchesToMeters(2.0), // elevator drum radius
+      LOWER_LIMIT, // min elevator height meters
+      UPPER_LIMIT, // max elevator height meters
+      true,
+      0,
+      0.01,
+      0.0);
+
+  // private DCMotorSim elevatorSim = new DCMotorSim(
+  // LinearSystemId.createDCMotorSystem(DCMotor.getFalcon500(1), 0.004,
+  // motorReduction),
+  // DCMotor.getFalcon500(1));
+
+  private final EncoderSim simEncoder = new EncoderSim(encoder);
+  private final PWMSim simMotor = new PWMSim(motor);
+
+  private final LoggedMechanism2d mech2d = new LoggedMechanism2d(20, 50, new Color8Bit(Color.kBlue));
+  private final LoggedMechanismRoot2d mech2dRoot = mech2d.getRoot("elevator root", 10, 0);
+  private final LoggedMechanismLigament2d elevatorMech2d = mech2dRoot.append(
+      new LoggedMechanismLigament2d("elevator", elevatorSim.getPositionMeters(), 90, 5, new Color8Bit(Color.kCoral)));
+
+  public ElevatorIOSim() {
+    // distance per pulse = (distance per revolution) / (pulses per revolution)
+    encoder.setDistancePerPulse(2.0 * Math.PI * (Units.inchesToMeters(2.0)) / 4096);
+
+    SmartDashboard.putData("elevator sim", mech2d);
+  }
+
+  @Override
+  public void updateInputs(ElevatorIOInputs inputs) {
```

Comment:
Is there a way to simulate sensors in this?

I'd love to see a way to set `inputs.elevatorSensor`